### ISSUE TITLE
Calculator with WinAppDriver 0.7

### ISSUE THREAD

**Issue by amitg25** [Microsoft/WinAppDriver#100]
I am trying to run Calculator App (java version) with WinAppDriver 0.5, but I see following issues with Appium.

When I declare my variables as IOSDriver, 
private static IOSDriver<IOSElement> CalculatorSession = null;
private static WebElement CalculatorResult = null;
with following capabilities

capabilities.setCapability("app", "Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
capabilities.setCapability("platformName", "Windows");
capabilities.setCapability("deviceName", "WindowsPC");

I keep getting iOS driver and exception
org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. Original error: Could not find path to Xcode by symlinks located in /var/db/xcode_select_link, or /usr/share/xcode-select/xcode_dir_link (WARNING: The server did not provide any stacktrace information)

Now if I change it to
 private static RemoteWebDriver CalculatorSession = null;
    private static RemoteWebElement CalculatorResult = null;

It starts working but now WinAppDriver fails to getText from screen. Following line fails
 final String xPath = "//*[@AutomationId='CalculatorResults']";
            String text = CalculatorSession.findElementByXPath(xPath).getText();

I tried changing xpath to different ways like
//windows[@name='Display is 7 ']; 
//*[@name='Display is 7 ']; 
but none worked.
however when I call  CalculatorSession.findElementByName("Seven").click();, this run successfully. 

Can someone suggest what's wrong in here? To me it looks like driver issue.

**Comment by ianceicys** [comment]
Can you please try the new .7 version? 
https://github.com/Microsoft/WinAppDriver/releases/tag/v0.7-beta
I've been able to get it working and have a demo here:
https://www.youtube.com/watch?v=XAJVpvaEchY

**Comment by amitg25** [comment]
Here are the logs

[Appium] Welcome to Appium v1.6.1
 [Appium] Appium REST http interface listener started on 0.0.0.0:4723
 [HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"app":"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App","platformName":"Windows","deviceName":"WindowsPC"}}
 [debug] [MJSONWP] Calling AppiumDriver.createSession() with args: [{"app":"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App","platformName":"Windows","deviceName":"WindowsPC"},null,null,null,null]
 [Appium] Creating new WindowsDriver session
 [Appium] Capabilities:
 [Appium] app: 'Microsoft.WindowsCalculator_8wekyb3d8bbwe!App'
 [Appium] platformName: 'Windows'
 [Appium] deviceName: 'WindowsPC'
 [BaseDriver] The following capabilities were provided, but are not recognized by appium: app.
 [BaseDriver] Session created with session id: d6a2889c-9021-40b4-bde1-f4035f1c710b
 [WinAppDriver] Verifying WinAppDriver is installed with correct checksum
 [debug] [WinAppDriver] Deleting WinAppDriver session
 [MJSONWP] Encountered internal error running command: Error: Could not verify WinAppDriver install; re-run install
 at WinAppDriver.start$ (lib/winappdriver.js:35:13)
 at tryCatch (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
 at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
 at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js💯21)
 at GeneratorFunctionPrototype.invoke (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
 [HTTP] <-- POST /wd/hub/session 500 93 ms - 190
 [HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"app":"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App","platformName":"iOS","deviceName":"WindowsPC"}}
 [debug] [MJSONWP] Calling AppiumDriver.createSession() with args: [{"app":"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App","platformName":"iOS","deviceName":"WindowsPC"},null,null,null,null,null,null]
 [Appium] Creating new IosDriver session
 [Appium] Capabilities:
 [Appium] app: 'Microsoft.WindowsCalculator_8wekyb3d8bbwe!App'
 [Appium] platformName: 'iOS'
 [Appium] deviceName: 'WindowsPC'
 [debug] [iOS] IosDriver version: 1.13.0
 [BaseDriver] Session created with session id: 615e7f47-a463-4cf3-80a1-0d357ce9a1c2
 [Xcode] Finding XcodePath by symlink because Error: Command 'xcode-select --print-path' errored out: Error: spawn xcode-select ENOENT
 at exports._errnoException (util.js:1008:11)
 at Process.ChildProcess._handle.onexit (internal/child_process.js:182:32)
 at onErrorNT (internal/child_process.js:348:16)
 at _combinedTickCallback (internal/process/next_tick.js:74:11)
 at process._tickCallback (internal/process/next_tick.js:98:9)
 [Xcode] Could not find path to Xcode by symlinks located in /var/db/xcode_select_link, or /usr/share/xcode-select/xcode_dir_link
 [debug] [iOS] Error: Could not find path to Xcode by symlinks located in /var/db/xcode_select_link, or /usr/share/xcode-select/xcode_dir_link
 at getPathFromSymlink$ (lib/xcode.js:79:9)
 at tryCatch (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
 at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
 at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js💯21)
 at GeneratorFunctionPrototype.invoke (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
 Error: Could not find path to Xcode by symlinks located in /var/db/xcode_select_link, or /usr/share/xcode-select/xcode_dir_link
 at getPathFromSymlink$ (lib/xcode.js:79:9)
 at tryCatch (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
 at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
 at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js💯21)
 at GeneratorFunctionPrototype.invoke (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
 [iOS] Could not determine Xcode version: Could not find path to Xcode by symlinks located in /var/db/xcode_select_link, or /usr/share/xcode-select/xcode_dir_link
 [MJSONWP] Encountered internal error running command: Error: Could not find path to Xcode by symlinks located in /var/db/xcode_select_link, or /usr/share/xcode-select/xcode_dir_link
 at getPathFromSymlink$ (lib/xcode.js:79:9)
 at tryCatch (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
 at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
 at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js💯21)
 at GeneratorFunctionPrototype.invoke (C:\Users\agupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
 [HTTP] <-- POST /wd/hub/session 500 170 ms - 257

**Comment by amitg25** [comment]
Hi ianceicys, I am using appium which by default install 0.5 beta version but when I upgrade manually to version 0.7, Appium refuse to find WinAppDriver and complain, it's not installed.

**Comment by amitg25** [comment]
Here are the logs

[Appium] Creating new WindowsDriver session
[Appium] Capabilities:
[Appium]   app: 'Microsoft.WindowsCalculator_8wekyb3d8bbwe!App'
[Appium]   platformName: 'Windows'
[Appium]   deviceName: 'WindowsPC'
[BaseDriver] The following capabilities were provided, but are not recognized by appium: app.
[BaseDriver] Session created with session id: a2a2ef15-ab16-47d9-8217-8c258df6c1c9
[WinAppDriver] Verifying WinAppDriver is installed with correct checksum
[debug] [WinAppDriver] Deleting WinAppDriver session
[MJSONWP] Encountered internal error running command: Error: Could not verify WinAppDriver install; re-run install
    at WinAppDriver.start$ (lib/winappdriver.js:35:13)
    at tryCatch (C:\Users\amgupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\amgupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\amgupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (C:\Users\amgupta\AppData\Roaming\npm\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
[HTTP] <-- POST /wd/hub/session 500 99 ms - 190

**Comment by amitg25** [comment]
Hi ianceicys, I stopped Appium and run the WinAppDriver version 0.7 with Calculator Sample app and It seems even sample is failing and unable to find element using XPath. Here is the log from console

==========================================
POST /session/8367C8D6-DDA1-42F3-8680-BE9DC6EA43F7/element HTTP/1.1
Accept: application/json, image/png
Content-Length: 67
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723

{"using":"xpath","value":"//Button[starts-with(@Name, \"Menu,\")]"}
HTTP/1.1 404 Not Found
Content-Length: 139
Content-Type: application/json

{"status":7,"value":{"error":"no such element","message":"An element could not be located on the page using the given search parameters."}}

**Comment by ianceicys** [comment]
@amitg25 hmmm...looking into the java version...I tested the C# version and that's working in .7 version. Can you see if the C# version is working on your machine? https://www.youtube.com/watch?v=XAJVpvaEchY

**Comment by amitg25** [comment]
Thanks a lot ianceicys for the video. I followed the video and run the test but even C# is failing with call findElementByXPath. I am on 0.7 build. I also confirmed that xPath is correct. Please check the attached 

![xpatherror](https://cloud.githubusercontent.com/assets/104231/20844090/064527ba-b873-11e6-87de-711221eb26f4.png)

**Comment by amitg25** [comment]
With further analysis, it seems only FindElementByXPath is failing, however findElementByName is works fine.

**Comment by amitg25** [comment]
I see an existing bug, look like the issue i am facing.

FindElementsByXPath should not throw NoSuchElementException  bugView all bug issues  
 #79 opened on Oct 20 by jneidlingerView open issues created by jneidlinger   v0.7  View all issues in this milestone

**Comment by amitg25** [comment]
Another one. https://github.com/Microsoft/WinAppDriver/issues/106

This bug is important to be fixed, it's kind of blocker .

**Comment by johleung** [comment]
I have the same issue as @amitg25 which I filed in #106. I'm using WIN10 build 10586, VS2015 update 3, Selenium 3.0, WinAppDriver 0.7. Sample c# calculator project. 

I downgraded Selenium to 2.53 and WinAPPDriver 0.6 with no luck. 
I have success on a dev VM - WIN10 build 14393, VS2015 update 3, Selenium 3.0, WinAppDriver 0.7.  
Note: my WIN10 build 10586 is a corporate build (locked down) so I'm not able to upgrade to later build.
